firefox-ios/Client/Frontend/Fakespot/FakespotViewController.swift

```diff
@@ -416,8 +416,7 @@ class FakespotViewController: UIViewController,
         case .onboarding:
             let view: FakespotOptInCardView = .build()
             viewModel.optInCardViewModel.dismissViewController = { [weak self] action in
-                store.dispatch(FakespotAction.setAppearanceTo(BoolValueContext(boolValue: false, windowUUID: windowUUID)))
-
+                self?.triggerDismiss()
```

Comment:
These changes result in the sidebar/modal not being restored when it should in fact be. As mentioned on the [issue](https://github.com/mozilla-mobile/firefox-ios/issues/19346#issuecomment-2056396171): on iPad the sidebar/modal should be restored/opened up unless the user turned off Fakespot or when they tapped outside the modal or closed it via the close or shopping icon buttons.